src/components/common/TmSessionExplore.vue

```diff
@@ -170,18 +178,24 @@ export default {
         return false
       }
     },
-    isANetworkAddress(param) {
-      const selectedNetwork = this.addressPrefixes.find(
-        ({ id }) => id === this.network
+    async selectNetworkByAddress(address) {
+      let selectedNetwork = this.addressPrefixes.filter(({ address_prefix }) =>
+        address.startsWith(address_prefix)
       )
       // handling query not loaded yet or failed
-      if (!selectedNetwork) return false
-
-      if (param.startsWith(selectedNetwork.address_prefix)) {
-        return true
+      if (!selectedNetwork) {
+        console.error("Connecting to the selected network failed")
+        return
+      }
+      // handling when there are both mainnet and testnet networks
+      if (selectedNetwork.length > 1) {
+        selectedNetwork = selectedNetwork.filter(({ id }) =>
+          this.testnet ? id.slice(-7) === `testnet` : id.slice(-7) === `mainnet`
```

Comment:
Networks have a testnet property so we don't need this funky string destructuring

Reply:
I like funky 🤖

Reply:
hahahah but we don't need complex code ^^
this also assumes that the ids of the networks always have `testnet` or something in it.